source/fundamentals/connection/tls.txt

```diff
@@ -0,0 +1,222 @@
+``MongoClientSettings``.
+
+.. code-block:: csharp
+
+   var cert = new X509Certificate2("client.pfx", "mySuperSecretPassword");
+
+   var settings = new MongoClientSettings
+   {
+      SslSettings = new SslSettings
+      {
+         ClientCertificates = new[] { cert }
+      },
+      UseTls = true
+   };
+
+.. note:: 
+
+   When loading a certificate with a password, the certificate object must contain a private 
+   key. If it doesn't, your certificate will not be passed to the server.
+
+Allow Insecure TLS
+------------------
+
+When TLS is enabled, the {+driver-short+} automatically verifies the certificate
+the server presents. When testing your code, you can disable certificate verification;
+this is known as *insecure TLS.* 
+
+When using insecure TLS, the only requirement is that the server present an X.509 
+certificate. The driver will accept a certificate even if any of the following are true: 
+
+- The hostname of the server and the subject name (or subject alternative name) 
+  on the certificate don't match.
+- The certificate is expired or not yet valid. 
+- The certificate doesn't have a trusted root certificate in the chain. 
+- The certificate purpose isn't valid for server identification.
+
+You can disable certificate verification in two different ways: using a property on a 
+``MongoClientSettings`` object or through a parameter in your connection string.
+
+.. tabs::
+
+   .. tab:: MongoClientSettings
+      :tabid: mongoclientsettings
+
+      To turn off automatic certificate verification with a ``MongoClientSettings`` 
+      object, set the ``AllowInsecureTls`` property to ``true``:
+
+      .. code-block:: csharp
+         :emphasize-lines: 4
+
+         var settings = new MongoClientSettings 
+         { 
+            UseTls = true,
+            AllowInsecureTls = true 
+         };
+         var client = new MongoClient(settings);
+      
+   .. tab:: Connection String
+      :tabid: connectionstring
+
+      To turn off automatic certificate verification using a connection string, 
+      assign the connection string parameter ``tlsInsecure`` a value of ``true``:
+
+      .. code-block:: csharp
+
+         var mongoClient = new MongoClient("mongodb://user1:password1@sample.host:27017?tls=true&tlsInsecure=true");
+      
+.. warning::
+
+   Always set this option to ``false`` in production. For security reasons, it's 
+   important that the server certificate is properly validated.
+
+.. _tls_configure-certificates:
+
+Check Certificate Revocation
+----------------------------
+
+When an X.509 certificate should no longer be trusted--for example, if its private key
+has been compromised--the certificate authority will revoke the certificate. 
+
+By default, the {+driver-short+} doesn't check whether a server's certificate has been
+revoked before it connects. You can enable revocation checking using either 
+``MongoClientSettings`` or the connection string.
```

Comment:
Question: Are there any common use cases where a user should be making sure they enable revocation checking? Or at least any that are worth calling out here?

Reply:
Waiting to hear back from the drivers team on this. If it's okay with you, I'd like to get this merged, and then we can come back to add in whatever we hear from them later.